src/main/java/org/apache/chemistry/opencmis/utils/FilePersistenceLoader.java

```diff
@@ -49,118 +52,13 @@ public boolean accept(File dir, String name) {
-            File folder, String folderId, FilenameFilter filenameFilter,
-            PersistenceManager persistenceManager) {
+        
         LOG.info("Scanning " + folder.getAbsolutePath());
-        // iterate through children
-        for (File child : folder.listFiles(filenameFilter)) {
-            // skip hidden files
-            if (child.isHidden()) continue;
-
-        	LOG.debug("Loading file " + child.getAbsolutePath());
-        	
-        	boolean toBeSaved = false;
-        	
-            StoredObject so = null;
-            if (child.isDirectory()) {
-                so = new FolderImpl(child.getName(), folderId);
-                so.setName(child.getName());
-                so.setId(persistenceManager.generateId());
-                File metadataFile = new File(child.getAbsolutePath()
-                        + SUFFIXE_METADATA);
-                
-                String relativePath = child.getAbsolutePath().replace(persistenceManager.getRootPath(), "");
-                
-                FolderImpl meta = (FolderImpl) persistenceManager
-                        .readCMISFromDisk(metadataFile);
-                if (meta != null) {
-                    // if metadata file exists then meta is the CMIS Object
-                    so = meta;
-                } else {
-                    LOG.warn("Missing metadata or malformed file for " + child.getAbsolutePath());
-                    so.setProperties(new LinkedHashMap<String, PropertyData<?>>());
-                    so.setTypeId(BaseTypeId.CMIS_FOLDER.value());
-                    so.setName(child.getName());
-                    toBeSaved = true;
-                }
-                Map<String, PropertyData<?>> properties = so.getProperties();
-                if (!properties.containsKey(PropertyIds.PATH) 
-                        || !properties.get(PropertyIds.PATH).getFirstValue().equals(relativePath)) {
-                    LOG.warn("Fixing cmis:path for " + relativePath);
-                    properties.put(PropertyIds.PATH, new PropertyStringImpl(PropertyIds.PATH, relativePath));
-                    //toBeSaved = true;
-                }
-                Folder fso = (Folder) so;
-                if (fso.getTypeId() == null) fso.setTypeId("cmis:folder");
-                if (!folderId.equals(fso.getParentId())) {
-                    LOG.warn("Fixing folder.parentId : setting " + folderId + " into cmis:parentId");
-                    fso.setParentId(folderId);
-                    //toBeSaved = true;
-                }
-                so.setRepositoryId(repositoryId);
-                so.setStore(store);
-            } else {
-                so = new DocumentImpl();
-                File metadataFile = new File(child.getAbsolutePath()
-                        + SUFFIXE_METADATA);
-                so.setId(persistenceManager.generateId());
+        
+        ExecutorService pool = Executors.newFixedThreadPool(100);
```

Comment:
I think it could be interesting to use an external settings to manage the number of threads.

The threads pool efficiency depends on CPU cores availability so if you do that you will able to tune the number of threads according to the capacity of the server.

https://engineering.zalando.com/posts/2019/04/how-to-set-an-ideal-thread-pool-size.html

Reply:
Right! OK for adding a new env variable. Stay tuned!

Reply:
fixed in https://github.com/MGDIS/lightweightCMISserver/pull/19/commits/376a754f4924fd9cb40afd236dbe19ef18037f93